### PR TITLE
Add support for LoongArch.

### PR DESCRIPTION
**Pull request by liuxiang88**
This commit is adding support for LoongArch architecture.

### CODE REVIEW

**Comment by selmf** [comment]
Thank you for the PR. I see no issue in merging this, but it would be great if you could ask the upstream project for the LZMA sdk (7zip) if they are interested in adding LoongArch support.

You can find the project here: https://sourceforge.net/p/sevenzip/patches/

I also need some general information like your name and an email to add you to the list of authors. Of course, if you'd prefer to contribute anonymous we can do that too.

**Comment by liuxiang88** [comment]
Thank you!
My name is liuxiang,Email is liuxiang@loongson.cn.

**Comment by selmf** [comment]
Ok, I have added you to AUTHORS and updated the changelog. Could you take a quick look and tell me if everything is correct?

**Comment by liuxiang88** [comment]
> Ok, I have added you to AUTHORS and updated the changelog. Could you take a quick look and tell me if everything is correct?

Thank you.I've already seen it and there's no problem.Is it time to merge my commit?

**Comment by selmf** [comment]
Merged, thank you for the contribution.

File: lzmasdk/CpuArch.h [hunk 1]
```diff
@@ -83,6 +83,14 @@ MY_CPU_LE_UNALIGN means that CPU is LITTLE ENDIAN and CPU supports unaligned mem
 #endif
 
 
+#if  defined(__loongarch64) \
+  || defined(__loongarch__) \
```
[the remark this answers is not in the document]
Comment (liuxiang88) [reply lzmasdk/CpuArch.h:87]: Yes,you are right.After the GCC version is updated, the definition changes.I'll make another commit to update.

[the remark this answers is not in the document]
Comment (zjiaz) [reply lzmasdk/CpuArch.h:87]: Okay. And I thought I have left the comment before the merge, but actually it was just pending in my timeline. So sorry for the late review.

**Pull request merged**